sdk/tests/js/tests/tex-image-and-sub-image-2d-with-canvas.js

```diff
@@ -195,13 +324,41 @@ function generateTest(internalFormat, pixelFormat, pixelType, prologue, resource
                       },
                       debug);
             } else {
+                var localRed   = semiTransparent ? semiTransparentRedColor : redColor;
+                var localGreen = semiTransparent ? semiTransparentGreenColor : greenColor;
+
+                // Allow a tolerance for premultiplication/unmultiplication, especially for texture
+                // formats with lower bit depths.
+                var tolerance = 0;
+                if (semiTransparent) {
+                    tolerance = 3;
+                    if (pixelType == 'UNSIGNED_SHORT_5_6_5' || internalFormat == 'RGB565') {
+                        tolerance = 6;
+                    } else if (pixelType == 'UNSIGNED_SHORT_4_4_4_4' || internalFormat == 'RGBA4') {
+                        tolerance = 9;
+                    } else if (pixelType == 'UNSIGNED_SHORT_5_5_5_1' || internalFormat == 'RGB5_A1') {
+                        // It's ambiguous whether the one bit of alpha should be treated as:
```

Comment:
I think this u8norm->u1norm conversion would fall under the normalized->float and float->normalized rule, applying them sequentially as u8norm->float->u1norm. In this case, this is specified in GLES 3.0.5 p10, and can actually always return either of the nearest values.
I do think we should test that 0/255->0/1 and 255/255->1/1 are preserved. (otherwise this format is useless)